kafka/consumer.go

```diff
@@ -308,25 +365,28 @@ func (c *Consumer) StoreMessage(m *Message) (storedOffsets []TopicPartition, err
 
 // Seek seeks the given topic partitions using the offset from the TopicPartition.
 //
-// The ignoredTimeoutMs parameter is ignored. Instead, this method blocks until
-// the fetcher state is updated for the given partition with the new offset.
-// This guarantees that no previously fetched messages for the old offset (or
-// fetch position) will be passed to the application once this call returns.
-// It will still take some time after the method returns until messages are
-// fetched at the new offset.
+// If timeoutMs is not 0 the call will wait this long for the
+// seek to be performed. If the timeout is reached the internal state
+// will be unknown and this function returns ErrTimedOut.
+// If timeoutMs is 0 it will initiate the seek but return
+// immediately without any error reporting (e.g., async).
 //
 // Seek() may only be used for partitions already being consumed
 // (through Assign() or implicitly through a self-rebalanced Subscribe()).
 // To set the starting offset it is preferred to use Assign() and provide
 // a starting offset for each partition.
 //
 // Returns an error on failure or nil otherwise.
-func (c *Consumer) Seek(partition TopicPartition, ignoredTimeoutMs int) error {
+func (c *Consumer) Seek(partition TopicPartition, timeoutMs int) error {
+	err := c.verifyClient()
+	if err != nil {
+		return err
+	}
 	rkt := c.handle.getRkt(*partition.Topic)
 	cErr := C.rd_kafka_seek(rkt,
 		C.int32_t(partition.Partition),
 		C.int64_t(partition.Offset),
-		C.int(-1))
+		C.int(timeoutMs))
```

Comment:
these changes are not related to this PR, should not be here

Reply:
Reverted.